### PR TITLE
Feat: VRAM and Speed optimization

### PR DESCRIPTION
**Pull request by y-prudent**
**Features**

- **`TorchFeatureExtractor`**: Wraps the forward pass in `torch.no_grad()` when `detach=True` to prevent gradient tracking.
- **`TorchFeatureExtractor` & `KerasFeatureExtractor`**: Introduce `numpy_concat` flag in `predict`, accumulating outputs as NumPy arrays rather than GPU tensors => significantly reduces VRAM usage for large datasets.
- **`TorchFeatureExtractor` & `KerasFeatureExtractor`**: Move all concatenations out of the batch loop, reducing concatenation complexity from O(N²) to O(N) for much faster execution.
- Extend feature-based OOD detection methods and React threshold computation to support `numpy_concat`.

**Note:** The Gram detector works with very large stats tensors. Running quantile and deviation on the CPU (NumPy) is too slow, but moving whole tensors to the GPU can cause out-of-memory errors. To balance speed and memory use, we now send only the stats for one layer and one class at a time to the GPU for quantile and deviation calculations.

**TODO**

- [ ] In feature-based OOD detection methods, maybe it could be even more optimized: At some points, we still convert large arrays to GPU tensors with `self.op.from_numpy` => could this be avoided to reduce VRAM peaks?

### CODE REVIEW

File: oodeel/methods/rmds.py [hunk 1]
```diff
@@ -184,6 +190,9 @@ def _fit_layer(
             val_scores : np.ndarray | None
                 Per-sample RMDS scores for an aggregator, or `None`.
         """
+        if isinstance(layer_features, np.ndarray):
+            layer_features = self.op.from_numpy(layer_features)
```
[the remark this answers is not in the document]
Comment (y-prudent) [reply oodeel/methods/rmds.py:194]: Same answer as for maha

File: oodeel/extractor/keras_feature_extractor.py [hunk 2]
```diff
@@ -313,17 +319,33 @@ def predict(
...
                     if logits is None
-                    else tf.concat([logits, logits_batch], axis=0)
+                    else (
+                        np.concatenate([logits, logits_batch], axis=0)
+                        if numpy_concat
+                        else tf.concat([logits, logits_batch], axis=0)
+                    )
                 )
                 # concatenate labels of current batch with previous batches
                 if contains_labels:
```
[the remark this answers is not in the document]
Comment (y-prudent) [reply oodeel/extractor/keras_feature_extractor.py:339]: Yes good point, I updated torch and keras extractors to integrate this remark, as well as maha, rmds and she

File: oodeel/methods/gram.py [hunk 3]
```diff
@@ -227,14 +240,18 @@ def _fit_to_dataset(
         n_layers = len(self.feature_extractor.feature_layers_id)
         self.postproc_fns = [self._stat] * n_layers
 
+        if self.aggregator is None and n_layers > 1:
+            self.aggregator = StdNormalizedAggregator()
+
         stats_all, info = self.feature_extractor.predict(
```
[the remark this answers is not in the document]
Comment (y-prudent) [reply oodeel/methods/gram.py:246]: These are not raw features because the `self._stat` postprocessing function is applied to the features
